Logging: Restore and enhance legacy logging docs (#135093)
elastic/kibana: docs/setup/settings.asciidoc

`elasticsearch.logQueries`::
deprecated:[7.12.0,"This setting is no longer used and will be removed in Kibana 8.0."]
Instead, configure the `elasticsearch.query` logger and set an approriate appender for the logger.
+
This is useful for seeing the query DSL generated by applications that
currently do not have an inspector, for example Timelion and Monitoring.
*Default: `false`*
+
The following example shows a full, valid `elasticsearch.query` logging configuration:
+
[source,yaml]
--------------------------------------------------------------------------------
logging:
  appenders:
    console_appender: <1>
      type: console <2>
      layout:
        type: pattern <3>
        highlight: true
  root:
    appenders: [default, console_appender]
    level: warn
  loggers:
    - name: elasticsearch.query <4>
      level: debug
      appenders: [console_appender]
--------------------------------------------------------------------------------

> <1> Appender names are entirely custom with the restriction that appender names must be unique. Appenders may be used for multiple loggers.

Author: I'm using the example as "soft" introduction to logging configuration.

`elasticsearch.logQueries`::
deprecated:[7.12.0,"This setting is no longer used and will be removed in Kibana 8.0."]
Instead, configure the `elasticsearch.query` logger and an appender for it.
+
This is useful for seeing the query DSL generated by applications that
currently do not have an inspector, for example Timelion and Monitoring.
*Default: `false`*
+
The following example shows a full, valid `elasticsearch.query` logging configuration:
+
[source,yaml]
--------------------------------------------------------------------------------
logging:
  appenders:
    console_appender: <1>
      type: console <2>
      layout:
        type: pattern <3>
        highlight: true
  root:
    appenders: [default, console_appender]
    level: warn
  loggers:
    - name: elasticsearch.query <4>
      level: debug

>       appenders: [console_appender]

Reviewer: nit: technically this isn't necessary for the example to run, but I guess it shows how you can change appenders per logger

Author: Adding an example is part of the "soft introduction" to the new logging system. It also shows how to specify an appender for a specific logger context and not fall back to the root logger config.

Maximum number of documents loaded by each shard to generate autocomplete
suggestions. This value must be a whole number greater than zero.
*Default: `"100000"`*

NOTE: To reload the <<logging-settings, logging settings>>, send a SIGHUP signal to {kib}.

> // deprecated logging configuration settings

Author: I've split the legacy and current "settings" into two grouped sections to make it clear what is deprecated and what current settings are/will be supported in 8.0

the `polling` method could be used enabling that option. *Default: `false`*

`logging.silent`::
deprecated:[7.13.0,This setting will be removed in Kibana 8.0.]
 Instead, set the root logger `level` to `off`
 +
Set the value of this setting to `true` to
suppress all logging output. *Default: `false`*

`logging.timezone`::
deprecated:[7.13.0,This setting will be removed in Kibana 8.0.]
Instead, use the {kibana-ref}/logging-service.html#pattern-layout[`pattern` appender layout] and specify a `%date` conversion pattern
 +
Set to the canonical time zone ID
(for example, `America/Los_Angeles`) to log events using that time zone.
For possible values, refer to
https://en.wikipedia.org/wiki/List_of_tz_database_time_zones[database time zones]. *Default: `UTC`*

[[logging-verbose]] `logging.verbose` {ess-icon}::
deprecated:[7.13.0,This setting will be removed in Kibana 8.0.]
Instead, set the root logger `level` to `debug`
 +
Set to `true` to log all events, including system usage information and all
requests. *Default: `false`*

[WARNING]
============================================================================

> While combining deprecated logging configuration with the current configuration, we strongly recommend switching over completely to avoid unforeseen consequences. If you use log rotation, combining deprecated with current configuration can lead to accidentally overwriting log files.

Author: Warns up front about potential consequences of mixing the two systems.

Maximum number of documents loaded by each shard to generate autocomplete
suggestions. This value must be a whole number greater than zero.
*Default: `"100000"`*

[WARNING]
============================================================================

> While {kib} supports both deprecated and current logging systems being enabled at the same time, mixing the configurations is not recommended. To avoid unforeseen consequences, rather switch over to the current configuration completely.

Reviewer: super nit
```suggestion
While {kib} supports both deprecated and current logging systems being enabled at the same time, mixing the configurations is not recommended. To avoid unforeseen consequences, we recommend switching over to the current configuration completely.
```

For more information, refer to the <<`logging-configuration-migration`, logging configuration migration guide>>
============================================================================

// deprecated logging configuration settings
[[logging-dest]] `logging.dest`::
 deprecated:[7.13.0,This setting will be removed in Kibana 8.0.]
 Instead, specify an appender `type`. Options are `console` for stdout or `file` for sending logs to file.
 +

>  Enables you to specify a file where Kibana stores log output. *Default: `stdout`*

Reviewer 2: NIT should we use `{kib}` instead?

`logging.silent`::
deprecated:[7.13.0,This setting will be removed in Kibana 8.0.]
 Instead, set the root logger `level` to `off`.
 +
Set the value of this setting to `true` to
suppress all logging output. *Default: `false`*

`logging.timezone`::
deprecated:[7.13.0,This setting will be removed in Kibana 8.0.]
Instead, use the {kibana-ref}/logging-service.html#pattern-layout[`pattern` appender layout] and specify a `%date` conversion pattern.
 +
Set to the canonical time zone ID
(for example, `America/Los_Angeles`) to log events using that time zone.
For possible values, refer to
https://en.wikipedia.org/wiki/List_of_tz_database_time_zones[database time zones]. *Default: `UTC`*

[[logging-verbose]] `logging.verbose` {ess-icon}::
deprecated:[7.13.0,This setting will be removed in Kibana 8.0.]
Instead, set the root logger `level` to `debug`.
 +
Set to `true` to log all events, including system usage information and all
requests. *Default: `false`*

NOTE: To reload the <<logging-settings, logging settings>>, send a SIGHUP signal to {kib}.

// new logging configuration settings
[[logging-root]] `logging.root`::

> The `root` logger has is a <<dedicated-loggers, dedicated logger>> and is pre-configured. The `root` logger logs at `info` level by default. If any other logger is configured, the `root` logger _must_ also be explicitly configured with an appropriate appender. 

Reviewer: > If any other logger is configured, the `root` logger _must_ also be explicitly configured with an appropriate appender.

Is that true? I think this is still a valid config:

```yaml
logging:
  root:
    level: warn
  loggers:
    - name: elasticsearch.query
      level: debug
```

Author: um, you're right, I've just tested it.

Author: ```suggestion
The `root` logger has is a <<dedicated-loggers, dedicated logger>> and is pre-configured. The `root` logger logs at `info` level by default. 
```

Author: Ah, now I remember what I was trying to say: "If any other logger is configured, the `root` logger _must_ also be explicitly defined/configured in `kibana.yml`". "with an appropriate appender" is superfluous.

[[logging-root-level]] `logging.root.level` {ess-icon}::

> Level at which a log record should be logged. Supported levels are: _all_, _fatal_, _error_, _warn_, _info_, _debug_, _trace_, _off_. Levels are ordered from _all_ (highest) to _off_ and a log record will be logged it its level is higher than or equal to the level of its logger, otherwise the log record is ignored. Use this value to <<change-overall-log-level,change the overall log level>>. Set to `all` to log all events, including system usage information and all requests. Set to `off` to silence all logs. *Default: `info`*. 

Reviewer: grammar nit:
```suggestion
Level at which a log record should be logged. Supported levels are: _all_, _fatal_, _error_, _warn_, _info_, _debug_, _trace_, _off_. Levels are ordered from _all_ (highest) to _off_, and a log record will be logged it its level is higher than or equal to the level of its logger, otherwise the log record is ignored. Use this value to <<change-overall-log-level,change the overall log level>>. Set to `all` to log all events, including system usage information and all requests. Set to `off` to silence all logs. *Default: `info`*. 
```

Author: The Oxford comma gets me every time!

+
The following example shows a full, valid `root` logging configuration that will log at the `debug` level:
+
[source,yaml]
--------------------------------------------------------------------------------

> logging:

Author: specifically addresses https://github.com/elastic/kibana/issues/134462
